### PR TITLE
Add system tests for legacy data types.

### PR DESCRIPTION
**Pull request by tseaver**
Toward #3029.

### CODE REVIEW

**Comment by tseaver** [comment]
Hold off on review:  the `use_legacy_sql` flag is set incorrectly.

**Comment by tseaver** [comment]
OK, I've pushed a version which flips the flag and drops the couldn't-possibly-work standard-SQL-only examples.

File: system_tests/bigquery.py [hunk 1]
```diff
@@ -479,6 +479,63 @@ def _job_done(instance):
         # raise an error, and that the job completed (in the `retry()`
         # above).
 
+    def test_sync_query_w_legacy_sql_types(self):
+        import datetime
```
Comment (dhermes) [inline system_tests/bigquery.py]: Feel free to import this at the global level

File: system_tests/bigquery.py [hunk 2]
```diff
@@ -479,6 +479,63 @@ def _job_done(instance):
...
+        from google.cloud._helpers import UTC
+        from google.cloud.bigquery._helpers import ArrayQueryParameter
+        from google.cloud.bigquery._helpers import ScalarQueryParameter
+        from google.cloud.bigquery._helpers import StructQueryParameter
+        naive = datetime.datetime(2016, 12, 5, 12, 41, 9)
+        stamp = "%s %s" % (naive.date().isoformat(), naive.time().isoformat())
```
Comment (dhermes) [inline system_tests/bigquery.py]: Single quote?

File: system_tests/bigquery.py [hunk 3]
```diff
@@ -479,6 +479,63 @@ def _job_done(instance):
...
+            },
+            {
+                'sql': 'SELECT CAST("%s" AS TIMESTAMP)' % (stamp,),
+                'expected': zoned,
+            },
+        ]
+        for example in EXAMPLES:
+            query = Config.CLIENT.run_sync_query(
+                example['sql'],
+                query_parameters=example.get('query_parameters', ()))
```
Comment (dhermes) [inline system_tests/bigquery.py]: None of the examples have `query_parameters`

Comment (tseaver) [inline system_tests/bigquery.py]: Good catch:  legacy SQL doesnt support them.

File: system_tests/bigquery.py (hunk 1 above)
Comment (tseaver) [inline system_tests/bigquery.py]: I'm allergic. :)

File: system_tests/bigquery.py (hunk 2 above)
Comment (tseaver) [inline system_tests/bigquery.py]: OK

File: system_tests/bigquery.py (hunk 3 above)
Comment (tseaver) [inline system_tests/bigquery.py]: 214046d

**Comment by dhermes** [comment]
LGTM, but ping me once CI is green so I can do a final pass?

**Comment by tseaver** [comment]
@dhermes CI is green.

**Pull request merged**